### PR TITLE
Run all tests and report failure at the end

### PR DESCRIPTION
**Pull request by phracek**
This commit runs all tests and report
the results at the end of the tests

Signed-off-by: Petr "Stone" Hracek <phracek@redhat.com>

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
@pkubatrh @fila43  PTAL. It should run all tests a report results at the end. THX

File: test/run [hunk 1]
```diff
@@ -16,7 +16,18 @@ test -n "${IMAGE_NAME-}" || { echo 'make sure $IMAGE_NAME is defined' && false ;
 test -n "${VERSION-}" || { echo 'make sure $VERSION is defined' && false; }
 test -n "${OS-}" || { echo 'make sure $OS is defined' && false; }
 
-test_exit=1
+test_short_summary=''
+TESTSUITE_RESULT=0
+
+TEST_LIST="\
+run_container_creation_tests
```
Comment (fila43) [inline test/run:23]: From my point of view is missing check_result inside run_container_creation_tests

File: test/run [hunk 2]
```diff
@@ -248,43 +271,69 @@ function run_tests() {
...
   local container_ip
   container_ip=$(get_container_ip $name)
   assert_login_access "$container_ip" "$PASS" true
+  check_result $?
   if [ -n "$PASS" ] ; then
     assert_login_access "$container_ip" "${PASS}_foo" false
+    check_result $?
   fi
   assert_local_access "$name"
+  check_result $?
```
Comment (fila43) [inline test/run:293]: if function assert_local_access fails "Success" is printed out is it correct?

File: test/run [hunk 3]
```diff
@@ -16,7 +16,18 @@ test -n "${IMAGE_NAME-}" || { echo 'make sure $IMAGE_NAME is defined' && false ;
...
+TESTSUITE_RESULT=0
+
+TEST_LIST="\
+run_container_creation_tests
+run_tests_no_root
+run_tests_no_pass
+run_tests_no_pass_altuid
+run_tests_no_root_altuid
+run_change_password_test
+run_doc_test
```
Comment (fila43) [inline test/run:29]: run_doc_test could return 1, but nothing checks it. I would add check_result into run_all_tests after $test_case

File: test/run [hunk 4]
```diff
@@ -248,43 +271,69 @@ function run_tests() {
...
+  DOCKER_ARGS="-u 12345" PASS=pass run_tests no_root_altuid
+}
 
-# Test the password change
-run_change_password_test
+function run_all_tests() {
+  for test_case in $TEST_SET; do
+    info "Running test $test_case ...."
+    TESTCASE_RESULT=0
+    $test_case
```
Comment (fila43) [inline test/run:323]:
```suggestion
    $test_case
    check_result $?
```

File: test/run (hunk 1 above)
Comment (phracek) [reply test/run:23]: Could point I have added `check_result` inside `try_image_invalid_combinations`.

File: test/run (hunk 2 above)
Comment (phracek) [reply test/run:293]: It can be printed. The result is 0. Success means for this test.

Comment (phracek) [reply test/run:293]: Definitely not, It has to be fixed.

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
Tests are passing. All issues were addressed. @fila43 Would you be willing to take a look at it one more time?

**Comment by fila43** [comment]
Thanks! now it looks great LGTM

**Pull request merged**